### PR TITLE
Re enroll

### PR DESCRIPTION
**Pull request by byteZorvin**
- This adds the reactivate and unblock functionality for a token
- A token that has been active can be only be deactivated
- Only Unknown token can be blocked
- Adds relevant testcases
image

### CODE REVIEW

File: tests/token_bridge_test.cairo [hunk 1]
```diff
@@ -339,3 +350,142 @@ fn disable_withdrwal_not_owner() {
     );
 }
 
+#[test]
+fn unblock_token_ok() {
```
Comment (apoorvsadana) [inline tests/token_bridge_test.cairo:354]: let's write a unit test for this

Comment (apoorvsadana) [reply tests/token_bridge_test.cairo:354]: for all below

File: tests/token_bridge_test.cairo [hunk 2]
```diff
@@ -339,3 +338,116 @@ fn disable_withdrwal_not_owner() {
...
+    let mut mock = mock_state_testing();
+    let usdc_address = deploy_erc20("usdc", "usdc");
```
Comment (apoorvsadana) [inline tests/token_bridge_test.cairo:344]: do we need to deploy a token?

File: tests/token_bridge_test.cairo [hunk 3]
```diff
@@ -339,3 +338,116 @@ fn disable_withdrwal_not_owner() {
...
+
+    // Setting the token active
+    let old_settings = mock.token_settings.read(usdc_address);
+    mock
+        .token_settings
+        .write(usdc_address, TokenSettings { token_status: TokenStatus::Blocked, ..old_settings });
```
Comment (apoorvsadana) [inline tests/token_bridge_test.cairo:350]: so we can directly write to storage in mocks?

Comment (byteZorvin) [reply tests/token_bridge_test.cairo:350]: Oh yes, found a way to do this. This import `use starknet_bridge::bridge::token_bridge::TokenBridge::__member_module_token_settings::InternalContractMemberStateTrait;` allows to do it

File: tests/token_bridge_test.cairo (hunk 2 above)
Comment (byteZorvin) [reply tests/token_bridge_test.cairo:344]: Oh ya makes sense, can mock an address to do so as well

Comment (apoorvsadana) [reply tests/token_bridge_test.cairo:344]: yep, makes sense to do it here

Comment (byteZorvin) [reply tests/token_bridge_test.cairo]: Updated ✅

**Pull request merged**